libs/entry-components/styles/modules/components/toggle/_generator.scss

```diff
@@ -22,24 +22,30 @@
 
 		@each $state in $enabled-states {
 			.mdc-switch:#{$state} {
-				&.mdc-switch--selected #{$handle-selector} {
-					background-color: $handle-on-background;
+				@if $handle-on-background {
+					&.mdc-switch--selected #{$handle-selector} {
+						background-color: $handle-on-background;
+					}
 				}
 
-				&:not(.mdc-switch--selected) #{$handle-selector} {
-					background-color: $handle-off-background;
+				@if $handle-off-background {
+					&:not(.mdc-switch--selected) #{$handle-selector} {
+						background-color: $handle-off-background;
+					}
 				}
 
-				.mdc-switch__track {
-					&::after {
-						background-color: -adjust-lightness($handle-on-background);
-					}
-			
-					&::before {
-						background-color: -adjust-lightness($handle-off-background);
+				@if $handle-on-background and $handle-off-background {
```

Comment:
Do we require both colors to be defined or is it enough just one for everything to work?
IMHO it's fine either way, but this logic is conflicting. First two ifs are allowing independent definition, but the last one requires both...

Reply:
We require both of them to this be able to work. Now we have only one condition and I moved it up.